### DISCUSSION TITLE
How to add tasks with schedules in an already running scheduler.

### DISCUSSION THREAD

**Discussion by suspiciousRaccoon** [taskiq-python/taskiq#275] [Q&A] [answered]
Hi!
This library looks really good but I'm unsure on how to add tasks with custom schedules to the scheduler. In celery you can do this with [`add_periodic_task()`](https://docs.celeryq.dev/en/stable/reference/celery.html#celery.Celery.add_periodic_task). I looked a bit at the source code and if I understood correctly the schedules are added when the task is discovered. Is this correct?.

**Comment by s3rius** [comment, marked as answer]
There is an undocumented way for doing this. You can call `schedule_by_*` methods on the task.

Methods are defined here: 

https://github.com/taskiq-python/taskiq/blob/d8237b80cedf686170bba5174fa3e060de5cf51d/taskiq/decor.py#L101
https://github.com/taskiq-python/taskiq/blob/d8237b80cedf686170bba5174fa3e060de5cf51d/taskiq/decor.py#L127
 
 These methods were in beta but now they can be officially added to docs. Docs will be updated soon.
 
 > schedules are added when the task is discovered
 
 This is true only for LabeledScheduleSource which operates on labels. But for schedule sources that support dynamic schedule declaration this is not true. As an example of such ScheuleSource you can take a look at `RedisScheduleSource`.
 
 https://github.com/taskiq-python/taskiq-redis/blob/09e835e4bfbd9e9a50ea4de440d23e6f2e4c5fd1/taskiq_redis/schedule_source.py#L12

**Reply by suspiciousRaccoon** [reply]
Thanks for the answer!
I didn't find mention of the redis scheduler in [Available schedule sources](https://taskiq-python.github.io/available-components/schedule-sources.html
) or in the package README.md's. Is it meant to be used? I tried using it and it seemed to work.

**Reply by s3rius** [reply]
Yes. It's ready to be used. The documentation was not yet updated.
